python/python/lance/torch/data.py

```diff
@@ -45,17 +46,25 @@ def _fsl_to_tensor(arr: pa.FixedSizeListArray, dimension: int) -> torch.Tensor:
 
 
 def _to_tensor(
-    batch: pa.RecordBatch,
+    batch: Union[pa.RecordBatch, Dict[str, pa.Array]],
     *,
     uint64_as_int64: bool = True,
     hf_converter: Optional[dict] = None,
 ) -> Union[dict[str, torch.Tensor], torch.Tensor]:
     """Convert a pyarrow RecordBatch to torch Tensor."""
     ret = {}
 
-    for col in batch.schema.names:
+    cols = (
+        batch.column_names if isinstance(batch, pa.RecordBatch) else list(batch.keys())
+    )
+    for col in cols:
         arr: pa.Array = batch[col]
 
+        if isinstance(arr, list) and arr and isinstance(arr[0], lance.BlobFile):
```

Comment:
Is there a way to check earlier? Like when constructing the loader?

Reply:
Yes we probably can. One way to do it is passing more parameters, but it makes user-specified `to_tensor_fn` more complicated.